src/inmanta/server/services/compilerservice.py

```diff
@@ -255,18 +254,20 @@ async def run(self, force_update: Optional[bool] = False) -> Tuple[bool, Optiona
             async def ensure_venv() -> Optional[data.Report]:
                 """
                 Ensure a venv is present at `venv_dir`.
-
-                :returns: `data.Report` reporting about the venv creation or None when the venv already exists.
                 """
-                python_path: str = PythonEnvironment.get_python_path_for_env_path(venv_dir)
-                if not os.path.exists(venv_dir) or not os.path.exists(python_path):
-                    LOGGER.info("Creating venv")
-                    return await self._run_compile_stage(
-                        name="Creating venv", cmd=[sys.executable, "-m", "venv", "--without-pip", venv_dir], cwd=project_dir
-                    )
-                else:
+                virtual_env = VirtualEnv(venv_dir)
+                if virtual_env.exists():
                     return None
```

Comment:
Because it would add a log line to the compile log saying `Creating venv`, while it didn't do anything at all. That could create confusion when debugging.